pkg/runtime/runtime.go

```diff
@@ -1261,13 +1261,14 @@ func (a *DaprRuntime) appendOrReplaceComponents(component components_v1alpha1.Co
 	}
 }
 
-func (a *DaprRuntime) extractComponentCategory(component components_v1alpha1.Component) ComponentCategory {
+func (a *DaprRuntime) extractComponentCategory(component components_v1alpha1.Component) (ComponentCategory, error) {
 	for _, category := range componentCategoriesNeedProcess {
 		if strings.HasPrefix(component.Spec.Type, fmt.Sprintf("%s.", category)) {
-			return category
+			return category, nil
 		}
 	}
-	return ""
+	// the category entered is incorrect, return error
+	return "", errors.Errorf("incorrect category %s", component.Spec.Type)
```

Comment:
`processComponentAndDependents` is the only method calling `extractComponentCategory`. I think we do not need to return error. Instead, `processComponentAndDependents` can check the return value is empty and then return the proper error.

what do you think?